exchanges/stream/websocket.go

```diff
@@ -347,34 +347,24 @@ func (w *Websocket) dataMonitor() {
 	w.Wg.Add(1)
 
 	go func() {
-		defer func() {
-			for {
-				// Bleeds data from the websocket connection if needed
-				select {
-				case <-w.DataHandler:
-				default:
-					w.setDataMonitorRunning(false)
-					w.Wg.Done()
-					return
-				}
-			}
-		}()
-
+		defer w.setDataMonitorRunning(false)
+		defer w.Wg.Done()
```

Comment:
`Done()` will execute before `setDataMonitorRunning()` might want to swap these just to be on the safe side.

Reply:
Good point. I doubt that it'd race, but I'm not sure. Best to be safe.

Fixed 61ac0e5e